### PR TITLE
Bugfix/code search tool

### PR DESCRIPTION
**Pull request by pranath-reddy**
### Summary :memo:
This PR fixes multiple issues in the Code Search Tool related to query storing, SDE search results, score normalization, and reranking logic. These changes improve result consistency when processing multiple queries.

### Details
- Fixed handling of the `query` field by explicitly attaching the query string to each result.  
- Corrected SDE search output by ensuring results are properly extended with associated queries.  
- Normalized CrossEncoder similarity scores using sigmoid function for stable values and Interpretability.  
- Updated reranking logic to rerank results per query and distribute `top_k` appropriately across multiple queries. 

### Bugfixes :bug: 
- Fixed missing or inconsistent `query` field in results.  
- Fixed SDE search output handling for multi-query scenarios.  
- Fixed unbounded CrossEncoder scores by normalizing them.  
- Fixed reranking to handle multi-query results correctly. 

### Checks
- [x] Tested Changes
- [ ] Stakeholder Approval

### CODE REVIEW

File: akd/tools/code_search.py [hunk 1]
```diff
@@ -214,8 +214,15 @@ async def _arun(
...
 
-        reranked = self._rerank_results(all_results, params.queries[0])[: params.top_k]
-        return self.output_schema(results=reranked, category="technology")
+        final_results = []
+        for query in params.queries:
+            query_results = [result for result in all_results if result.query == query]
+            reranked = self._rerank_results(query_results, query)[
+                : (params.top_k) // len(params.queries)
+            ]
+            final_results.extend(reranked)
```
Comment (NISH1001) [inline akd/tools/code_search.py]: Can this be made concurent? Normally asyncio handles it if we `gather(*tasks)`. I think it should be concurrrent as long as it doesn't impose any memory issue during reranking (which generates vector i guess). Does having concurrency impose memory issue if search results is too many?

Comment (NISH1001) [reply akd/tools/code_search.py]: Or should we concate the queries and rerank at just one time? I am not sure. You mentioned this multiple reranking leading to worser performance?

File: akd/tools/code_search.py [hunk 2]
```diff
@@ -230,6 +237,7 @@ def _rerank_results(
 
             # Get similarity scores from CrossEncoder
             scores = self.reranker_model.predict(pairs)
+            scores = 1 / (1 + np.exp(-scores))
```
Comment (NISH1001) [inline akd/tools/code_search.py:251]: Oh interesting. Does this help?

File: akd/tools/code_search.py [hunk 3]
```diff
@@ -511,6 +519,8 @@ async def _arun(
                     remove_embedding_column=self.config.remove_embedding_column,
                 )
                 if results:
+                    for result in results:
+                        result["query"] = query
                     all_results_data.extend(results)
```
Comment (NISH1001) [inline akd/tools/code_search.py:535]: Do we need if condition?

File: akd/tools/code_search.py (hunk 1 above)
Comment (pranath-reddy) [reply akd/tools/code_search.py]: I'll make it concurrent, but at the tool level we shouldn't rerank all queries together, right? We might use the tool with a list of independent queries as well, not just the reformulated ones. We can rerank them all togther at agent level.

File: akd/tools/code_search.py (hunk 3 above)
Comment (pranath-reddy) [reply akd/tools/code_search.py:535]: Just a safety check

File: akd/tools/code_search.py (hunk 2 above)
Comment (pranath-reddy) [reply akd/tools/code_search.py:251]: No, the relative order remains same, but `predict()` was returning raw logits so just wanted to make the scores more readable.

File: akd/tools/code_search.py (hunk 1 above)
Comment (NISH1001) [reply akd/tools/code_search.py]: We can make concurrent

Comment (pranath-reddy) [reply akd/tools/code_search.py]: Done

**Comment by NISH1001** [comment]
@pranath-reddy 2 tests are failing. Can you retry the tests again? The `test_end_to_end_workflow_with_report_output` should pass I guess as it's intermittent.

**Review by NISH1001 — APPROVED** [review]
